### PR TITLE
Improvements to plugin lifecycle #42

### PR DESCRIPTION
**Pull request by Badya**
Fix for issue #42

### CODE REVIEW

File: AspectJ-gradle/src/main/kotlin/com/archinamon/api/AspectJCompileTask.kt [hunk 1]
```diff
@@ -53,10 +53,10 @@ internal open class AspectJCompileTask: AbstractCompile() {
...
 
             val options = mutableMapOf(
-                Pair("overwrite", true),
-                Pair("group", "build"),
-                Pair("description", "Compile .aj source files into java .class with meta instructions"),
-                Pair("type", AspectJCompileTask::class.java)
+                    Pair("overwrite", true),
+                    Pair("group", "build"),
+                    Pair("description", "Compile .aj source files into java .class with meta instructions"),
+                    Pair("type", AspectJCompileTask::class.java)
```
Comment (Archinamon) [inline AspectJ-gradle/src/main/kotlin/com/archinamon/api/AspectJCompileTask.kt]: Let's avoid code-style changes :)

File: AspectJ-gradle/src/main/kotlin/com/archinamon/api/AspectJCompileTask.kt [hunk 2]
```diff
@@ -67,9 +67,6 @@ internal open class AspectJCompileTask: AbstractCompile() {
             task.aspectJWeaver = AspectJWeaver(project)
 
             task.source(sources)
-            task.classpath = classpath()
-
-            findCompiledAspectsInClasspath(task, config.includeAspectsFromJar)
```
Comment (Archinamon) [inline AspectJ-gradle/src/main/kotlin/com/archinamon/api/AspectJCompileTask.kt]: We could have no javac step (when using pure kotlin/groovy without java sources), so in this case we will have empty classpath. Not good.

File: AspectJ-gradle/src/main/kotlin/com/archinamon/plugin/PluginSetup.kt [hunk 3]
```diff
@@ -18,7 +18,6 @@ import org.gradle.api.plugins.PluginContainer
 
 internal fun configProject(project: Project, config: AndroidConfig, settings: AspectJExtension) {
     project.repositories.mavenCentral()
-    project.dependencies.add("compile", "org.aspectj:aspectjrt:${settings.ajc}")
```
Comment (Archinamon) [inline AspectJ-gradle/src/main/kotlin/com/archinamon/plugin/PluginSetup.kt]: Described in comment to issue.

File: AspectJ-gradle/src/main/kotlin/com/archinamon/api/AspectJCompileTask.kt (hunk 1 above)
Comment (Badya) [inline AspectJ-gradle/src/main/kotlin/com/archinamon/api/AspectJCompileTask.kt]: I just use IDEA's `Code -> Reformat code` all the time =)

**Pull request merged**